packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/drift/connection/connection_io.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/flutter_provider_interfaces/cached_events_path_provider.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:drift/drift.dart';
+import 'package:drift/isolate.dart';
+import 'package:drift/native.dart';
+import 'package:path/path.dart' as p;
+
+// Code adapted from:
+// https://github.com/simolus3/drift/blob/develop/examples/app/lib/database/connection/native.dart
+
+final AmplifyLogger _logger = AmplifyLogger.category(Category.analytics);
+
+/// {@template amplify_analytics_pinpoint_dart.connection_io}
+/// Obtains a database connection for running drift in a Dart VM.
+///
+/// The [NativeDatabase] from drift will synchronously use sqlite3's C APIs.
+/// To move synchronous database work off the main thread, we use a
+/// [DriftIsolate], which can run queries in a background isolate under the
+/// hood.
+/// {@endtemplate}
+QueryExecutor connect(CachedEventsPathProvider? pathProvider) {
+  return DatabaseConnection.delayed(
+    Future.sync(() async {
+      var dir = '';
```

Comment:
Yeah this is a tricky case..

It's only nullable because web doesn't need this class but it's required otherwise.  

Feasibly if there was a web only implementation, it doesn't make any sense to provide pathProvider that's why it's nullable.

I think the best approach would be to throw an exception.  If someone doesn't pass the pathProvider it'll throw and they'll know to fix it.